frontend/src/metabase/dashboard/selectors.ts

```diff
@@ -301,13 +303,44 @@ export const getParameterTarget = createSelector(
   },
 );
 
+export const getQuestions = (state: State) => {
+  const dashboard = getDashboard(state);
+
+  if (!dashboard) {
+    return [];
+  }
+
+  const dashcardIds = dashboard.dashcards;
+
+  const questionsById = dashcardIds.reduce<Record<DashCardId, Question>>(
+    (acc, dashcardId) => {
+      const dashcard = getDashCardById(state, dashcardId);
+
+      const question = getQuestionByDashcardMemo(state, { dashcard });
+
+      acc[dashcard.id] = question;
+
+      return acc;
+    },
+    {},
+  );
+
+  return questionsById;
+};
+
 export const getParameters = createSelector(
-  [getDashboardComplete, getMetadata],
-  (dashboard, metadata) => {
+  [getDashboardComplete, getMetadata, getQuestions],
+  (dashboard, metadata, questions) => {
```

Comment:
Not sure if it makes sense after all the performance fixes, but this selector doesn't really need to be "subscribed" to an entire dashboard object, we only care about `dashboard.parameters` and `parameter_mappings` in dashcards

Reply:
I'll add a separate task for it